app/assets/javascripts/discourse/app/components/discovery/filter-navigation.gjs

```diff
@@ -1,40 +1,86 @@
 import Component from "@glimmer/component";
 import { tracked } from "@glimmer/tracking";
 import { Input } from "@ember/component";
+import { fn } from "@ember/helper";
 import { on } from "@ember/modifier";
 import { action } from "@ember/object";
 import didInsert from "@ember/render-modifiers/modifiers/did-insert";
-import { next } from "@ember/runloop";
+import { cancel, later, next } from "@ember/runloop";
 import { service } from "@ember/service";
-import { and } from "truth-helpers";
+import { and, eq } from "truth-helpers";
 import BulkSelectToggle from "discourse/components/bulk-select-toggle";
 import DButton from "discourse/components/d-button";
-import FilterTips from "discourse/components/discovery/filter-tips";
+import DropdownMenu from "discourse/components/dropdown-menu";
 import PluginOutlet from "discourse/components/plugin-outlet";
 import bodyClass from "discourse/helpers/body-class";
+import concatClass from "discourse/helpers/concat-class";
 import icon from "discourse/helpers/d-icon";
+import element from "discourse/helpers/element";
 import lazyHash from "discourse/helpers/lazy-hash";
+import { ajax } from "discourse/lib/ajax";
 import discourseDebounce from "discourse/lib/debounce";
 import { bind } from "discourse/lib/decorators";
 import { resettableTracked } from "discourse/lib/tracked-tools";
 import { i18n } from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+
+const MAX_RESULTS = 20;
 
 export default class DiscoveryFilterNavigation extends Component {
   @service site;
 
   @tracked copyIcon = "link";
   @tracked copyClass = "btn-default";
   @tracked inputElement = null;
+  @tracked showTips = false;
+  @tracked searchResults = [];
   @resettableTracked newQueryString = this.args.queryString;
 
+  activeFilter = null;
+  searchTimer = null;
+  handleBlurTimer = null;
+
+  @tracked _selectedIndex = -1;
```

Comment:
I think we should stop using these `_xxxx`. How is `_selectedIndex` more private that `handleBlurTimer` for example?